home work 22/js/index.js

```diff
@@ -0,0 +1,224 @@
+  }
+
+  #generateFood() {
+    let randomRow = Math.floor(Math.random() * this.gridCount);
+    let randomCell = Math.floor(Math.random() * this.gridCount);
+
+    while (
+      this.#snake.some(
+        (snakeData) =>
+          snakeData.cell === randomCell && snakeData.row === randomRow
+      )
+    ) {
+      randomRow = Math.floor(Math.random() * this.gridCount);
+      randomCell = Math.floor(Math.random() * this.gridCount);
+    }
+
+    this.#food = { row: randomRow, cell: randomCell };
+    let foodCell = this.#findByCoords(this.#food);
+    foodCell.innerHTML = '<img src="img/apple.png" alt="food">';
+    console.log(foodCell);
+  }
+
+  #start() {
+    this.#snake = this.#buildSnake(
+      Math.floor(this.gridCount / 2),
+      Math.floor(this.gridCount / 2)
+    );
+
+    this.direction = DR.LEFT;
+
+    this.#generateFood();
+    this.#speed = +this.#controls.speed.value;
+    this.#messageContainer.innerHTML = "Welcome to Snake !";
+
+    document.querySelector("#snake-start-game").style.display = "none";
+    document.querySelector("#snake-end-game").style.display = "inline-block";
+
+    this.#process = setInterval(() => {
+      let { cell, row } = this.#snake[0];
+
+      switch (this.direction) {
+        case DR.LEFT:
+          {
+            cell = cell === 0 ? this.gridCount - 1 : cell - 1;
+            this.#snake.unshift({ cell, row });
+          }
+          break;
+
+        case DR.RIGHT:
+          {
+            cell = cell === this.gridCount - 1 ? 0 : cell + 1;
+            this.#snake.unshift({ cell, row });
+          }
+          break;
+
+        case DR.UP:
+          {
+            row = row === 0 ? this.gridCount - 1 : row - 1;
+            this.#snake.unshift({ cell, row });
+          }
+          break;
+
+        case DR.DOWN:
+          {
+            row = row === this.gridCount - 1 ? 0 : row + 1;
+            this.#snake.unshift({ cell, row });
+          }
+          break;
+      }
+
+      this.#clear();
+      this.#update();
+    }, this.#speed);
+  }
+
+  #clearFood() {
+    let foodCell = this.#findByCoords(this.#food);
+    foodCell.innerHTML = "";
+  }
+
+  #checkHasFoodEaten() {
+    if (
+      this.#snake[0].cell === this.#food.cell &&
+      this.#snake[0].row === this.#food.row
+    ) {
+      this.#clearFood();
+      this.#generateFood();
+      this.#score += 1;
+      document.querySelector("#snake-score b").innerHTML = this.#score;
+    } else {
+      this.#snake.pop();
+    }
+  }
+
+  #checkOnTailCrash() {
+    for (let i = 1; i < this.#snake.length; i++) {
+      if (
+        this.#snake[0].cell === this.#snake[i].cell &&
+        this.#snake[0].row === this.#snake[i].row
+      ) {
+        clearInterval(this.#process);
+        this.#messageContainer.innerHTML =
+          "Game Over. Your score is " + this.#score;
+        document.querySelector("#snake-end-game").style.display =
+          "inline-block";
```

Comment:
That's why you need to call the `end` method. That method uses the same code so ( as an example ) you can pass the message to its

Reply:
If the game has over the end btn should be hidden and start btn is shown